src/apps/ENERGY.cpp

```diff
@@ -97,5 +97,17 @@ void ENERGY::tearDown(VariantID vid)
   deallocData(m_vnewc);
 }
 
+size_t ENERGY::getBytesPerRep() const
+{
+  // some branches are never taken due to the nature of the initialization of delvc
+  // the additional reads and writes that would be done if those branches were taken are noted in the comments
+  return (1*sizeof(Real_type) + 5*sizeof(Real_type)) * getRunSize() +
+         (1*sizeof(Real_type) + 1/*+7*/*sizeof(Real_type)) * getRunSize() +
+         (1*sizeof(Real_type) + 6*sizeof(Real_type)) * getRunSize() +
+         (1*sizeof(Real_type) + 2*sizeof(Real_type)) * getRunSize() +
+         (1*sizeof(Real_type) + 7/*+5*/*sizeof(Real_type)) * getRunSize() +
+         (0/*+1*/*sizeof(Real_type) + 1/*+7*/*sizeof(Real_type)) * getRunSize() ;
```

Comment:
It looks like there is something wrong with the expressions on these lines; e.g., '7/*+5*/*sizeof(...'

Reply:
This is the way it is because some of the memory accesses are conditional on the value of the arrays and the way we initialize the values they are always between 0 and1 so some branches are never taken. 7/\*+5\*/ would be 12 if the other branch was taken.